community/kernel/src/main/java/org/neo4j/kernel/builtinprocs/SchemaProcedure.java

```diff
@@ -62,82 +63,90 @@ public GraphResult buildSchemaGraph()
         final Map<String,NodeImpl> nodes = new HashMap<>();
         final Map<String,Set<RelationshipImpl>> relationships = new HashMap<>();
 
-        ReadOperations readOperations = kernelTransaction.acquireStatement().readOperations();
-        StatementTokenNameLookup statementTokenNameLookup = new StatementTokenNameLookup( readOperations );
-
-        try ( Transaction transaction = graphDatabaseAPI.beginTx(); )
+        try ( Statement statement = kernelTransaction.acquireStatement() )
         {
-            // add all labelsInDatabase
-            ResourceIterator<Label> labelsInDatabase = graphDatabaseAPI.getAllLabelsInUse().iterator();
-            while ( labelsInDatabase.hasNext() )
-            {
-                Label label = labelsInDatabase.next();
-                Map<String,Object> properties = new HashMap<>();
-
-                Iterator<IndexDescriptor> indexDescriptorIterator =
-                        readOperations.indexesGetForLabel( readOperations.labelGetForName( label.name() ) );
-                ArrayList<String> indexes = new ArrayList<>();
-                while ( indexDescriptorIterator.hasNext() )
-                {
-                    indexes.add( statementTokenNameLookup
-                            .propertyKeyGetName( indexDescriptorIterator.next().getPropertyKeyId() ) );
-                }
-                properties.put( "indexes", indexes );
-
-                Iterator<NodePropertyConstraint> nodePropertyConstraintIterator =
-                        readOperations.constraintsGetForLabel( readOperations.labelGetForName( label.name() ) );
-                ArrayList<String> constraints = new ArrayList<>();
-                while ( nodePropertyConstraintIterator.hasNext() )
-                {
-                    constraints
-                            .add( nodePropertyConstraintIterator.next().userDescription( statementTokenNameLookup ) );
-                }
-                properties.put( "constraints", constraints );
-
-                getOrCreateLabel( label.name(), properties, nodes );
-            }
-
-            //add all relationships
+            ReadOperations readOperations = statement.readOperations();
+            StatementTokenNameLookup statementTokenNameLookup = new StatementTokenNameLookup( readOperations );
 
-            Iterator<RelationshipType> relationshipTypeIterator =
-                    graphDatabaseAPI.getAllRelationshipTypesInUse().iterator();
-            while ( relationshipTypeIterator.hasNext() )
+            try ( Transaction transaction = graphDatabaseAPI.beginTx(); )
```

Comment:
Do we need this inner transaction? We are already in a transaction, so this will be a dummy.